README.md

```diff
@@ -20,9 +20,54 @@ Via Composer
 $ composer require php-http/httplug-bundle
 ```
 
+Enable the bundle in your kernel:
+ 
+``` php
+<?php
+// app/AppKernel.php
+
+public function registerBundles()
+{
+    $bundles = array(
+        // ...
+        new Http\HttplugBundle\HttplugBundle(),
+    );
+}
+```
 
 ## Usage
 
+The usage documentation is split into two parts. First we explain how to configure the bundle in an application. The second part is for developing reusable Symfony bundles that depend on an HTTP client defined by the Httplug interface.
+
+### Use in applications
+
+This bundle provides 3 services: 
+
+* `httplug.client` a service that provides the `Http\Client\HttpClient`
+* `httplug.message_factory` a service that provides the `Http\Message\MessageFactory`
+* `httplug.uri_factory` a service that provides the `Http\Message\UriFactory`
+
+These services are always an alias to another service. You can specify your own service or leave the default, which is the same name with `.default` appended. The default services in turn use the service discovery mechanism to provide the best available implementation. You can specify a class for each of the default services to use instead of discovery. 
+
+If you need to customize the service with decorators, e.g. to add authentication headers, decorate the default service and configure the main alias to your decorating service name.
+
+```yaml
+httplug:
+    main_alias:
+        client: httplug.client.default
+        message_factory: httplug.message_factory.default
+        uri_factory: httplug.uri_factory.default
+    classes:
+        client: ~ # uses discovery if not specified
```

Comment:
Hum, maybe we can have a class for each adapter with can inject/send a NodeConfiguration ? So we would now how to configure each adapter.